### PR TITLE
feat(core): add tags-operations and tags-operations-split output modes

### PR DESCRIPTION
**Pull request by jakiestfu**
## Summary

Adds two new output modes that emit **one file per operation** nested under a per-tag directory, for use cases that want finer-grained generated files than `tags`/`tags-split` (one file per tag) provide:

- **`tags-operations`** — one file per operation containing both types and runtime hook/client code, e.g. `<build>/pets/get-pet.ts`.
- **`tags-operations-split`** — two files per operation: `<build>/pets/get-pet.ts` (runtime) + `<build>/pets/get-pet.schemas.ts` (the component schemas that operation references, resolved transitively).

Both modes also write:
- A per-tag barrel (`<build>/pets/index.ts`) that re-exports every operation's runtime file, so existing `import from '<build>/pets'`-style imports keep working.
- A per-tag shared helper file (`<build>/pets/pets.helpers.ts`) for boilerplate that today gets inlined once per tag (query-key helpers, `AwaitedInput`/`SecondParameter` type aliases, HTTP status code unions, etc.) — imported by name into each operation file instead of duplicated into every one.

### Client scope

These modes are scoped to clients that already compile each operation to a fully standalone function/hook: `react-query`, `svelte-query`, `vue-query`, `swr`, and `fetch`. Axios (and zod) wrap every operation in a single per-tag factory function (`export const getPets = (axiosInstance) => { ...; return { getPet, listPets } }`), and Angular/Hono group operations into a class/service/handler — none of these have an existing per-operation code path, so attempting to support them would mean changing their generated output shape. Configuring an unsupported client with either new mode throws a clear error pointing at `tags`/`tags-split` instead.

## Test plan

- [x] New unit tests: `packages/core/src/writers/tags-operations-mode.test.ts`, `tags-operations-split-mode.test.ts` (17 tests) covering per-operation file layout (dash-case filenames), the per-tag barrel, the shared helper file (both type-only and runtime-value imports, filtered to names each operation actually uses), transitive schema resolution, the global schemas file, per-operation mock files, the aggregate mock index across multiple tags, and the client guard.
- [x] Full existing writer suite (`packages/core/src/writers/**`) passes with no regressions — 185 tests.
- [x] Full `@orval/core`, `@orval/mock`, and `orval` package test suites pass (aside from 3 pre-existing, environment-dependent failures in `resolve-version.test.ts` unrelated to this change, reproducible on `master`).
- [x] End-to-end generation against the petstore spec with both new modes, output typechecked clean against `@tanstack/react-query`.

## Summary by CodeRabbit

* **New Features**
  * Added `tags-operations` and `tags-operations-split` output modes.
  * Generates runtime operation files grouped by tag (plus optional tag helper modules), with per-operation mocks and schema modules (including transitive schema dependencies).
  * Supports per-tag and tag-independent mock index/barrel generation, including MSW handler exports.
  * Adds client compatibility validations for these modes.
* **Bug Fixes**
  * Prevents ambiguous/duplicate workspace re-exports via updated index barrel behavior.
* **Tests**
  * Added writer tests covering filesystem outputs, mocks/schemas/barrels, and unsupported client errors.
* **Chores**
  * Improved faker array-item mock scoping to include per-operation isolation in these modes.

### CODE REVIEW

**Comment by melloware** [comment]
This feature seems great but the PR checks are failing?

File: packages/core/src/writers/tags-operations-mode.ts [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+                );
+                return ext === OutputMockType.MSW
+                  ? `export { get${pascal(opName)}Mock } from '${localMockPath}'\n`
+                  : `export * from '${localMockPath}'\n`;
+              })
+              .join('');
+            await writeGeneratedFile(indexPath, indexContent);
+            mockIndexPaths.push(indexPath);
+          }
+        }
```
[the remark this answers is not in the document]
Comment (jakiestfu) [reply packages/core/src/writers/tags-operations-mode.ts]: Good catch — fixed in 7bc7dfd14. Hoisted `mockIndexEntries` to function scope (now carrying each operation's tag), accumulated across all tags, and moved the write to a single pass after `Promise.all` resolves, sorted by tag then operation for idempotent output. Applied to both `tags-operations` and `tags-operations-split`, mirroring the existing pattern in `split-tags-mode.ts`. Added a regression test with two tagged mocks + `indexMockFiles: true` asserting the single root mock index lists operations from both tags.

**Comment by jakiestfu** [comment]
@melloware I think this is ready to go. I understand its a bit more code than the usual PR but considering its for 2 entirely new writers, I think it's reasonable. I will be doing some further audits to make sure nothing is exorbitant, but this seems ready to go.

We've tested it on our spec file at Turo and were able to fix a few bugs with the agents initial round of code changes, primarily around pathing and import cleanups in resulting files.

**Comment by melloware** [comment]
Excellent i am waiting for the build to pass

**Comment by melloware** [comment]
@jakiestfu can tyou review the latest Code Rabbit comments?

File: packages/core/src/writers/tags-operations-mode.ts [hunk 2]
```diff
@@ -0,0 +1,479 @@
...
+          const localMockPath = upath.joinSafe(
+            './',
+            tag,
+            opName + '.' + ext + mockImportExtension,
+          );
+          return ext === OutputMockType.MSW
+            ? `export { get${pascal(opName)}Mock } from '${localMockPath}'\n`
+            : `export * from '${localMockPath}'\n`;
+        })
+        .join('');
```
[the remark this answers is not in the document]
Comment (jakiestfu) [reply packages/core/src/writers/tags-operations-mode.ts:468]: Confirmed with actual repro (not the shell-approximated pascal/kebab in the analysis script) — this repo's real `pascal`/`kebab` round-trip losslessly for digit/acronym cases like `getV2Pet`/`get2Pet`, so that specific failure mode doesn't reproduce. But the underlying bug is real and worse than described: the index re-derives the export name as `get${pascal(opName)}Mock`, while the actual exported symbol (from `packages/mock/src/msw/index.ts`) is `mockOutput.implementation.handlerName`, i.e. `get${pascal(operationName)}MockHandler` — wrong for *every* operation, not just edge cases (missing `Handler" suffix). Fixed in f03013495 by carrying `handlerName` through the accumulated mock-index entry instead of re-deriving it. Tightened the regression test to assert the exact `export { <handlerName> }` line.

File: packages/core/src/writers/tags-operations-split-mode.test.ts [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+
+    const paths = await writeTagsOperationsSplitMode({
+      ...props,
+      needSchema: false,
+    });
+
+    expect(
+      fs.existsSync(path.join(tmpDir, 'health', 'getHealth.schemas.ts')),
+    ).toBe(false);
+    expect(paths.some((p) => p.endsWith('.schemas.ts'))).toBe(false);
```
[the remark this answers is not in the document]
Comment (jakiestfu) [reply packages/core/src/writers/tags-operations-split-mode.test.ts:209]: Confirmed and fixed in f03013495 — updated the assertion to `get-health.schemas.ts`.

File: packages/orval/src/write-specs.ts [hunk 4]
```diff
@@ -417,19 +419,50 @@ function getImplementationPathsForIndex(
...
     targetInfo.dirname,
     `${targetInfo.filename}.schemas${output.fileExtension}`,
   );
 
-  return excludeFilePath(paths, defaultSiblingSchemas);
+  return paths.filter(
+    (p) =>
+      getComparableFilePath(p) === getComparableFilePath(rootBarrel) ||
+      getComparableFilePath(p) === getComparableFilePath(globalSchemas),
+  );
```
[the remark this answers is not in the document]
Comment (jakiestfu) [reply packages/orval/src/write-specs.ts:465]: Confirmed and fixed in f03013495. `getImplementationPathsForIndex`'s tags-operations narrowing is correct for computing the barrel's re-export specifiers, but it was also propagating into `implementationPaths` -> `paths`, which feeds `afterAllFilesWrite` and `runFormatter` — silently dropping every per-tag/operation/helper/schema/mock file from both whenever `output.workspace` is set. Scoped the fix so the narrowed list is used only for the `imports` computation; the reassignment now keeps the full path list (minus the index file itself, matching the existing non-tags-operations behavior).

File: packages/core/src/writers/target-tags-operations.ts [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+  );
+  const exported = implementation
+    .replaceAll(TYPE_DECLARATION_RE, (whole, existingExport: string) =>
+      existingExport ? whole : `export ${whole}`,
+    )
+    .replaceAll(VALUE_DECLARATION_RE, (whole, existingExport: string) =>
+      existingExport ? whole : `export ${whole}`,
+    );
+  return { implementation: exported, typeNames, valueNames };
+}
```
[the remark this answers is not in the document]
Comment (jakiestfu) [reply packages/core/src/writers/target-tags-operations.ts:93]: Investigated but not applying a fix — this doesn't reproduce for any of the 5 clients `tags-operations`/`tags-operations-split` currently support. I traced every header/footer builder that's actually reachable (`packages/query/src/index.ts` for react-query/vue-query/svelte-query, `packages/fetch/src/index.ts`, `packages/swr/src/index.ts`): none emit top-level `interface`/`enum`/`class`/`let`/`var` in their header/footer implementation or `sharedTypes` — only `type` and `const` declarations (`AwaitedInput`, `SecondParameter`, `WITH_QUERY_KEY_HELPER`, the fetch `HTTPStatusCode*` shared types). Angular does emit some of those constructs, but Angular is explicitly unsupported for these modes (`assertClientSupportsTagsOperations` throws). Happy to revisit if a specific client's actual output can be shown to hit this — but adding speculative regex handling for constructs no supported client emits felt like the wrong tradeoff here.

File: packages/core/src/writers/tags-operations-mode.test.ts [hunk 6]
```diff
@@ -0,0 +1,451 @@
...
+    // `createSplitModeOperation` fixture), not a name re-derived from the
+    // kebab-cased operation filename (which previously produced e.g.
+    // `getListPetsMock` instead of the actual exported `mockHandler`).
+    const mockIndex = fs.readFileSync(mockIndexPath, 'utf8');
+    expect(mockIndex).toContain(
+      "export { mockHandler } from './pets/list-pets.msw'",
+    );
+    expect(mockIndex).toContain(
+      "export { mockHandler } from './health/get-health.msw'",
+    );
```
[the remark this answers is not in the document]
Comment (jakiestfu) [reply packages/core/src/writers/tags-operations-mode.test.ts:416]: Confirmed — although unique `operationId`s guarantee unique `handlerName`s by default, a user-supplied `override.operationName` function can map multiple operations to the same name (same pattern as the #3685 type-collision fix), which would collide here too. Fixed in 31b7a1ae5: each re-export is now aliased as `${pascal(tag)}${pascal(opName)}MockHandler`, derived from the entry's `{tag, opName}` dedup key (always unique), instead of re-exporting `handlerName` verbatim. Updated the regression test's assertions to the aliased names.

**Review by melloware — APPROVED** [review]
